### PR TITLE
continuation of PR #8161 Display error notification when attempting to purchase invalid amount of gems  - fixes #8145

### LINKED ISSUE #8145: buying zero gems for a gift with PayPal produces InternalServerError
On the gem gift screen, it's possible to attempt to purchase gems as a gift for another player without first entering the number of gems (i.e., to try to give zero gems). This produces an ugly error message when using PayPal (below). For Amazon, it allows you to click through to the Amazon site and for Stripe, it allows you to enter your credit card details, but both of those results are also wrong for a value of zero gems.

**The desired behaviour is that you should not be able to click on the "purchase" button until you have entered a number of gems.**

Steps to reproduce for the PayPal case (reported by Laurens van Riel, be51405a-b3c0-41f6-8f99-3e6af962d99f, in the Report a Bug guild):

**Start**:
On the page "Send gift to <username>" I got to this page from my tasks page, by clicking on a party member and then clicking the gift button.

**Actions**:
- Leaving the amount of gems on standard value=0
- Selecting the option "Purchase"
- Clicking the PayPal button.

**Result**:
White page with the text
`{"success":false,"error":"InternalServerError","message":"An unexpected error occurred."}`

### PR DESCRIPTION
**Pull request by TheHollidayInn**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #8145

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

A continuation of: https://github.com/HabitRPG/habitica/pull/8161. This also adds back sever checks for low gem amounts with unit tests now that we have abstracted the payments.

Credit goes to UUID: 5abc818d-baeb-4951-a66f-d6c1dba21305

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

File: website/common/locales/en/groups.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   "community": "<a target='_blank' href='http://habitica.wikia.com/wiki/Special:Forum'>Community Forum</a>",
   "dataTool": "Data Display Tool",
   "resources": "Resources",
-  "askQuestionNewbiesGuild": "Ask a Question (Habitica Help guild)",
+  "askQuestionNewbiesGuild": "Ask a Question (Newbies Guild)",
```
Comment (SabreCat) [inline website/common/locales/en/groups.json]: There seem to be a bunch of unrelated edits to this string file, some of them (such as this) clearly incorrect. Mistake resolving a merge conflict?

File: website/server/libs/amazonPayments.js [hunk 2]
```diff
@@ -98,6 +98,9 @@ api.checkout = async function checkout (options = {}) {
 
   if (gift) {
     if (gift.type === this.constants.GIFT_TYPE_GEMS) {
+      if (gift.gems.amount <= 0) {
+        throw new BadRequest(i18n.t('badAmountOfGemsToPurchase'));
```
Comment (SabreCat) [inline website/server/libs/amazonPayments.js:102]: We've been wanting not to have bare errors use translated strings IIRC--should this error message come from an API errors lib instead?

**Review by SabreCat — CHANGES_REQUESTED** [review]
Pretty minor things remaining here. One locale file needing some chunks reverted, one question that may not need to be resolved at this time.

File: website/server/libs/amazonPayments.js (hunk 2 above)
Comment (TheHollidayInn) [reply website/server/libs/amazonPayments.js:102]: Actually, we are using this on the front end as well, so maybe we just keep it in group translations for now?

**Review by SabreCat — APPROVED** [review]

**Pull request merged**